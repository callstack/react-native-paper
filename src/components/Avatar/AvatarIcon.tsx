import * as React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';

import { useInternalTheme } from '../../core/theming';
import { Colors } from '../../styles/themes/tokens';
import type { ThemeProp } from '../../types';
import getContrastingColor from '../../utils/getContrastingColor';
import Icon, { IconSource } from '../Icon';

const defaultSize = 64;

export type Props = React.ComponentPropsWithRef<typeof View> & {
  /**
   * Icon to display for the `Avatar`.
   */
  icon: IconSource;
  /**
   * Size of the avatar.
   */
  size?: number;
  /**
   * Custom color for the icon.
   */
  color?: string;
  style?: StyleProp<ViewStyle>;
  /**
   * @optional
   */
  theme?: ThemeProp;
};

/**
 * Avatars can be used to represent people in a graphical way.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { Avatar } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <Avatar.Icon size={24} icon="folder" />
 * );
 * ```
 */
const Avatar = ({
  icon,
  size = defaultSize,
  style,
  theme: themeOverrides,
  ...rest
}: Props) => {
  const {
    colors: { primary },
  } = useInternalTheme(themeOverrides);
  const { backgroundColor = primary, ...restStyle } =
    StyleSheet.flatten(style) || {};
  const textColor =
    rest.color ??
    getContrastingColor(
      backgroundColor,
      Colors.primary100,
      'rgba(0, 0, 0, .54)'
    );

  return (
    <View
      style={[
        {
          width: size,
          height: size,
          borderRadius: size / 2,
          backgroundColor,
        },
        styles.container,
        restStyle,
      ]}
      {...rest}
    >
      <Icon source={icon} color={textColor} size={size * 0.6} />
    </View>
  );
};

Avatar.displayName = 'Avatar.Icon';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default Avatar;
