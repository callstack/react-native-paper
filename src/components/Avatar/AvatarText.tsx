import * as React from 'react';
import {
  StyleProp,
  StyleSheet,
  TextStyle,
  useWindowDimensions,
  View,
  ViewStyle,
} from 'react-native';

import { useInternalTheme } from '../../core/theming';
import { Colors } from '../../styles/themes/tokens';
import type { ThemeProp } from '../../types';
import getContrastingColor from '../../utils/getContrastingColor';
import Text from '../Typography/Text';

const defaultSize = 64;

export type Props = React.ComponentPropsWithRef<typeof View> & {
  /**
   * Initials to show as the text in the `Avatar`.
   */
  label: string;
  /**
   * Size of the avatar.
   */
  size?: number;
  /**
   * Custom color for the text.
   */
  color?: string;
  /**
   * Style for text container
   */
  style?: StyleProp<ViewStyle>;
  /**
   * Style for the title.
   */
  labelStyle?: StyleProp<TextStyle>;
  /**
   * Specifies the largest possible scale a text font can reach.
   */
  maxFontSizeMultiplier?: number;
  /**
   * @optional
   */
  theme?: ThemeProp;
};

/**
 * Avatars can be used to represent people in a graphical way.
 *
 * ## Usage
 * ```js
 * import * as React from 'react';
 * import { Avatar } from 'react-native-paper';
 *
 * const MyComponent = () => (
 *   <Avatar.Text size={24} label="XD" />
 * );
 * ```
 */
const AvatarText = ({
  label,
  size = defaultSize,
  style,
  labelStyle,
  color: customColor,
  theme: themeOverrides,
  maxFontSizeMultiplier,
  ...rest
}: Props) => {
  const {
    colors: { primary },
  } = useInternalTheme(themeOverrides);
  const { backgroundColor = primary, ...restStyle } =
    StyleSheet.flatten(style) || {};
  const textColor =
    customColor ??
    getContrastingColor(
      backgroundColor,
      Colors.primary100,
      'rgba(0, 0, 0, .54)'
    );
  const { fontScale } = useWindowDimensions();

  return (
    <View
      style={[
        {
          width: size,
          height: size,
          borderRadius: size / 2,
          backgroundColor,
        },
        styles.container,
        restStyle,
      ]}
      {...rest}
    >
      <Text
        style={[
          styles.text,
          {
            color: textColor,
            fontSize: size / 2,
            lineHeight: size / fontScale,
          },
          labelStyle,
        ]}
        numberOfLines={1}
        maxFontSizeMultiplier={maxFontSizeMultiplier}
      >
        {label}
      </Text>
    </View>
  );
};

AvatarText.displayName = 'Avatar.Text';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    textAlign: 'center',
    textAlignVertical: 'center',
  },
});

export default AvatarText;
