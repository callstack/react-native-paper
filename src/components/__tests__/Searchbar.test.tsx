import * as React from 'react';
import { Animated } from 'react-native';

import { act, fireEvent, render } from '@testing-library/react-native';

import * as Avatar from '../Avatar/Avatar';
import Searchbar from '../Searchbar';

it('renders with placeholder', () => {
  const tree = render(<Searchbar placeholder="Search" value="" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders with text', () => {
  const tree = render(
    <Searchbar placeholder="Search" value="query" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('activity indicator snapshot test', () => {
  const tree = render(<Searchbar loading={true} value="" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders with ActivityIndicator', () => {
  const tree = render(<Searchbar loading={true} value="" />);

  expect(tree.getByTestId('activity-indicator')).toBeTruthy();
});

it('renders without ActivityIndicator', () => {
  const { getByTestId } = render(<Searchbar loading={false} value="" />);

  expect(() => getByTestId('activity-indicator')).toThrow();
});

it('render icon with custom ripple color', () => {
  const { getByTestId } = render(
    <Searchbar testID="search-bar" value={''} rippleColor="purple" />
  );

  const iconContainer = getByTestId('search-bar-icon-container').props.children;
  expect(iconContainer.props.rippleColor).toBe('purple');
});

it('renders clear icon with custom color', () => {
  const { getByTestId } = render(
    <Searchbar testID="search-bar" value="value" iconColor="purple" />
  );

  const iconComponent = getByTestId('search-bar-icon-wrapper').props.children;

  expect(iconComponent.props.iconColor).toBe('purple');
});

it('renders clear icon wrapper, which can be the target of touch events, if search has value', () => {
  const { getByTestId } = render(
    <Searchbar testID="search-bar" value="value" />
  );

  expect(getByTestId('search-bar-icon-wrapper').props.pointerEvents).toBe(
    'auto'
  );
});

it('renders clear icon wrapper, which is never target of touch events, if search has no value', () => {
  const { getByTestId } = render(<Searchbar testID="search-bar" value="" />);

  expect(getByTestId('search-bar-icon-wrapper').props.pointerEvents).toBe(
    'none'
  );
});

it('animated value changes correctly', () => {
  const value = new Animated.Value(1);
  const { getByTestId } = render(
    <Searchbar
      testID="search-bar"
      value=""
      style={[{ transform: [{ scale: value }] }]}
    />
  );
  expect(getByTestId('search-bar-container-outer-layer')).toHaveStyle({
    transform: [{ scale: 1 }],
  });

  Animated.timing(value, {
    toValue: 1.5,
    useNativeDriver: false,
    duration: 200,
  }).start();

  act(() => {
    jest.advanceTimersByTime(200);
  });
  expect(getByTestId('search-bar-container-outer-layer')).toHaveStyle({
    transform: [{ scale: 1.5 }],
  });
});

it('defines onClearIconPress action and checks if it is called when close button is pressed', () => {
  const onClearIconPressMock = jest.fn();
  const { getByTestId } = render(
    <Searchbar
      testID="search-bar"
      value="value"
      onClearIconPress={onClearIconPressMock}
    />
  );
  const iconComponent = getByTestId('search-bar-icon-wrapper').props.children;

  fireEvent(iconComponent, 'onPress');
  expect(onClearIconPressMock).toHaveBeenCalledTimes(1);
});

it('renders clear icon wrapper, with appropriate style for v3', () => {
  const { getByTestId, update } = render(
    <Searchbar testID="search-bar" value="" />
  );

  expect(getByTestId('search-bar-icon-wrapper')).toHaveStyle({
    position: 'absolute',
    right: 0,
    marginLeft: 16,
  });

  update(
    <Searchbar
      testID="search-bar"
      value=""
      right={() => <Avatar.Text label="AB" size={30} />}
    />
  );

  expect(getByTestId('search-bar-icon-wrapper')).toHaveStyle({
    display: 'none',
  });
});

it('render clear icon with custom ripple color', () => {
  const { getByTestId } = render(
    <Searchbar testID="search-bar" value={''} rippleColor="purple" />
  );

  const clearIconContainer = getByTestId('search-bar-clear-icon-container')
    .props.children;
  expect(clearIconContainer.props.rippleColor).toBe('purple');
});

it('renders trailering icon when mode is set to "bar"', () => {
  const { getByTestId } = render(
    <Searchbar
      testID="search-bar"
      value={''}
      traileringIcon={'microphone'}
      mode="bar"
    />
  );

  expect(getByTestId('search-bar-trailering-icon')).toBeTruthy();
});

it('renders trailering icon with press functionality', () => {
  const onTraileringIconPressMock = jest.fn();

  const { getByTestId } = render(
    <Searchbar
      testID="search-bar"
      value={''}
      traileringIcon={'microphone'}
      onTraileringIconPress={onTraileringIconPressMock}
      mode="bar"
    />
  );

  fireEvent(getByTestId('search-bar-trailering-icon'), 'onPress');
  expect(onTraileringIconPressMock).toHaveBeenCalledTimes(1);
});

it('renders trailering icon with custom ripple colors', () => {
  const { getByTestId } = render(
    <Searchbar
      testID="search-bar"
      value={''}
      traileringRippleColor={'purple'}
      traileringIcon={'microphone'}
      mode="bar"
    />
  );

  const traileringIconContainer = getByTestId(
    'search-bar-trailering-icon-container'
  ).props.children;
  expect(traileringIconContainer.props.rippleColor).toBe('purple');
});

it('renders clear icon instead of trailering icon', () => {
  const { getByTestId, update, queryByTestId } = render(
    <Searchbar
      testID="search-bar"
      value={''}
      traileringIcon={'microphone'}
      mode="bar"
    />
  );

  expect(getByTestId('search-bar-trailering-icon')).toBeTruthy();

  update(
    <Searchbar
      testID="search-bar"
      value={'test'}
      traileringIcon={'microphone'}
      mode="bar"
    />
  );

  expect(queryByTestId('search-bar-trailering-icon')).toBeNull();
  expect(getByTestId('search-bar-icon-wrapper')).toBeTruthy();
});

it('renders searchbar in "view" mode', () => {
  const { getByTestId } = render(
    <Searchbar testID="search-bar" value={''} mode="view" />
  );

  expect(getByTestId('search-bar-container')).toHaveStyle({ borderRadius: 0 });
});
