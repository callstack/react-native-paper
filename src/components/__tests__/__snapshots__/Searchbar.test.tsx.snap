// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity indicator snapshot test 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(238, 232, 244)",
      "borderRadius": 28,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="search-bar-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(238, 232, 244)",
        "borderRadius": 28,
        "flex": undefined,
        "flexDirection": "row",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="search-bar-container"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "height": 40,
          "margin": 6,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 40,
        }
      }
      testID="search-bar-icon-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="search-bar-icon-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel="search"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                undefined,
              ],
            ]
          }
          testID="search-bar-icon"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 24,
                },
                [
                  {
                    "lineHeight": 24,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
              ]
            }
          >
            magnify
          </Text>
        </View>
      </View>
    </View>
    <TextInput
      accessibilityRole="search"
      keyboardAppearance="light"
      placeholder=""
      placeholderTextColor="rgba(28, 27, 31, 1)"
      returnKeyType="search"
      selectionColor="rgba(103, 80, 164, 1)"
      style={
        [
          {
            "alignSelf": "stretch",
            "flex": 1,
            "fontSize": 18,
            "minWidth": 0,
            "paddingLeft": 8,
          },
          {
            "color": "rgba(73, 69, 79, 1)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 0,
            "textAlign": "left",
          },
          {
            "minHeight": 56,
            "paddingLeft": 0,
          },
          undefined,
        ]
      }
      testID="search-bar"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessibilityRole="progressbar"
      accessibilityState={
        {
          "busy": true,
        }
      }
      accessible={true}
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "marginHorizontal": 16,
          },
        ]
      }
      testID="activity-indicator"
    >
      <View
        collapsable={false}
        style={
          {
            "height": 24,
            "opacity": 1,
            "width": 24,
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 24,
                "transform": [
                  {
                    "rotate": "45deg",
                  },
                ],
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                [
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  },
                  null,
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "transform": [
                      {
                        "translateY": 0,
                      },
                      {
                        "rotate": "-165deg",
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 12,
                      "overflow": "hidden",
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "borderColor": "rgba(103, 80, 164, 1)",
                        "borderRadius": 12,
                        "borderWidth": 2.4,
                        "height": 24,
                        "width": 24,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            [
              {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 24,
                "transform": [
                  {
                    "rotate": "45deg",
                  },
                ],
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                [
                  {
                    "height": 12,
                    "overflow": "hidden",
                    "width": 24,
                  },
                  {
                    "top": 12,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "transform": [
                      {
                        "translateY": -12,
                      },
                      {
                        "rotate": "345deg",
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 12,
                      "overflow": "hidden",
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "borderColor": "rgba(103, 80, 164, 1)",
                        "borderRadius": 12,
                        "borderWidth": 2.4,
                        "height": 24,
                        "width": 24,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with placeholder 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(238, 232, 244)",
      "borderRadius": 28,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="search-bar-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(238, 232, 244)",
        "borderRadius": 28,
        "flex": undefined,
        "flexDirection": "row",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="search-bar-container"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "height": 40,
          "margin": 6,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 40,
        }
      }
      testID="search-bar-icon-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="search-bar-icon-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel="search"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                undefined,
              ],
            ]
          }
          testID="search-bar-icon"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 24,
                },
                [
                  {
                    "lineHeight": 24,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
              ]
            }
          >
            magnify
          </Text>
        </View>
      </View>
    </View>
    <TextInput
      accessibilityRole="search"
      keyboardAppearance="light"
      placeholder="Search"
      placeholderTextColor="rgba(28, 27, 31, 1)"
      returnKeyType="search"
      selectionColor="rgba(103, 80, 164, 1)"
      style={
        [
          {
            "alignSelf": "stretch",
            "flex": 1,
            "fontSize": 18,
            "minWidth": 0,
            "paddingLeft": 8,
          },
          {
            "color": "rgba(73, 69, 79, 1)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 0,
            "textAlign": "left",
          },
          {
            "minHeight": 56,
            "paddingLeft": 0,
          },
          undefined,
        ]
      }
      testID="search-bar"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      pointerEvents="none"
      style={
        [
          {
            "marginLeft": 16,
            "position": "absolute",
            "right": 0,
          },
          false,
        ]
      }
      testID="search-bar-icon-wrapper"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="search-bar-clear-icon-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="search-bar-clear-icon-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel="clear"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ],
              ]
            }
            testID="search-bar-clear-icon"
          >
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0)",
                    "fontSize": 24,
                  },
                  [
                    {
                      "lineHeight": 24,
                      "transform": [
                        {
                          "scaleX": 1,
                        },
                      ],
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ],
                ]
              }
            >
              close
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with text 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgb(238, 232, 244)",
      "borderRadius": 28,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="search-bar-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgb(238, 232, 244)",
        "borderRadius": 28,
        "flex": undefined,
        "flexDirection": "row",
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="search-bar-container"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "height": 40,
          "margin": 6,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": 40,
        }
      }
      testID="search-bar-icon-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="search-bar-icon-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel="search"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                undefined,
              ],
            ]
          }
          testID="search-bar-icon"
        >
          <Text
            accessibilityElementsHidden={true}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 24,
                },
                [
                  {
                    "lineHeight": 24,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
              ]
            }
          >
            magnify
          </Text>
        </View>
      </View>
    </View>
    <TextInput
      accessibilityRole="search"
      keyboardAppearance="light"
      placeholder="Search"
      placeholderTextColor="rgba(28, 27, 31, 1)"
      returnKeyType="search"
      selectionColor="rgba(103, 80, 164, 1)"
      style={
        [
          {
            "alignSelf": "stretch",
            "flex": 1,
            "fontSize": 18,
            "minWidth": 0,
            "paddingLeft": 8,
          },
          {
            "color": "rgba(73, 69, 79, 1)",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "400",
            "letterSpacing": 0.15,
            "lineHeight": 0,
            "textAlign": "left",
          },
          {
            "minHeight": 56,
            "paddingLeft": 0,
          },
          undefined,
        ]
      }
      testID="search-bar"
      underlineColorAndroid="transparent"
      value="query"
    />
    <View
      pointerEvents="auto"
      style={
        [
          false,
          false,
        ]
      }
      testID="search-bar-icon-wrapper"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="search-bar-clear-icon-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="search-bar-clear-icon-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel="clear"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ],
              ]
            }
            testID="search-bar-clear-icon"
          >
            <Text
              accessibilityElementsHidden={true}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                [
                  {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontSize": 24,
                  },
                  [
                    {
                      "lineHeight": 24,
                      "transform": [
                        {
                          "scaleX": 1,
                        },
                      ],
                    },
                    {
                      "backgroundColor": "transparent",
                    },
                  ],
                ]
              }
            >
              close
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
